addons/MyHome/MyHome.lua

```diff
@@ -0,0 +1,96 @@
+
+    * Redistributions of source code must retain the above copyright
+    notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+    notice, this list of conditions and the following disclaimer in the
+    documentation and/or other materials provided with the distribution.
+    * Neither the name of MyHome nor the
+    names of its contributors may be used to endorse or promote products
+    derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL from20020516 BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.]]
+
+_addon.name = 'MyHome'
+_addon.author = 'from20020516'
+_addon.version = '1.0'
+_addon.commands = {'myhome','mh','warp'}
+
+require('logger')
+extdata = require('extdata')
+
+lang = string.lower(windower.ffxi.get_info().language)
+item_info = {
+    [1]={id=28540,japanese='デジョンリング',english='"Warp Ring"',slot=13},
+    [2]={id=17040,japanese='デジョンカジェル',english='"Warp Cudgel"',slot=0},
+    [3]={id=4181,japanese='呪符デジョン',english='"Instant Warp"'}}
+
+function search_item()
+    local item_array = {}
+    local bags = {0,8,10,11,12} --inventory,wardrobe1-4
+    local get_items = windower.ffxi.get_items
+    for i=1,#bags do
+        for _,item in ipairs(get_items(bags[i])) do
+            if item.id > 0 then
+                item_array[item.id] = item
+                item_array[item.id].bag = bags[i]
+            end
+        end
+    end
+    for index,stats in pairs(item_info) do
+        local item = item_array[stats.id]
+        local set_equip = windower.ffxi.set_equip
+        if item then
+            ext = extdata.decode(item)
+            local enchant = ext.type == 'Enchanted Equipment'
+            local recast = enchant and ext.charges_remaining > 0 and math.max(ext.next_use_time+18000-os.time(),0)
+            local usable = recast and recast == 0
+            log(stats[lang],usable and '' or recast..' sec recast.')
+            if usable or ext.type == 'General' then
+                if enchant and item.status ~= 5 then --not equipped
+                    set_equip(item.slot,stats.slot,item.bag)
+                    log_flag = true
+                    repeat --waiting cast delay
+                        coroutine.sleep(1)
```

Comment:
Why not sleep once for the entire delay, instead of looping? You can get the delay from ext data, right?

Reply:
Probably, the ffxi client detects usable enchanted gear every 'tick' (time change event)
```lua
ext.usable == delay+math.random(0,3)
```
of course I could write something like `coroutine.sleep(delay+3)`,
but users use warp high-frequently resulting in a loss of user experience.